### PR TITLE
Add start and end dates to challenge

### PR DESCRIPTION
**Pull request by msmolens**
- Update challenge model and endpoints
- Add challenge tests

Note: introduces dependency on [iso8601](https://pypi.python.org/pypi/iso8601), a minimal module that parses ISO 8601 date strings into datetime objects.

### CODE REVIEW

File: server/models/challenge.py [hunk 1]
```diff
@@ -18,6 +18,7 @@
 ###############################################################################
 
 import datetime
+from iso8601 import iso8601
```
Comment (zachmullen) [inline server/models/challenge.py]: @msmolens Is there a reason we can't just use `dateutil.parser.parse` from python's standard library?

Comment (msmolens) [inline server/models/challenge.py]: `dateutil.parse.parse` is actually in the [python-dateutil](https://pypi.python.org/pypi/python-dateutil/) package, not the standard library. The standard library offers `datetime.strptime()`, but that would require explicitly handling many date formats if we want any flexibility in input. Using the iso8601 module enables robust parsing of a standard date/time format.

Comment (zachmullen) [inline server/models/challenge.py]: Hm, not sure why I thought dateutil was standard. Let's use that package instead, I can see us wanting to support a wide variety of date formats.

**Comment by msmolens** [comment]
Converted from iso8601 to python-dateutil for date/time parsing based on https://github.com/girder/challenge/pull/31#discussion_r44450475.

**Comment by zachmullen** [comment]
This looks good for the challenge. Now let's add the same fields and behavior to the phase resource as well.

**Comment by msmolens** [comment]
Added start and end date to phase.

**Comment by zachmullen** [comment]
Great job, LGTM

**Pull request merged**